docs/proposals/update_replace_functions.md

```diff
@@ -0,0 +1,76 @@
+# Proposal: Improve the replacement functions
+
+_Owners_: @daneshk @kalaiyarasiganeshalingam  
+_Reviewers_: @daneshk  
+_Created_: 2022/03/11   
+_Updated_: 2022/03/11  
+_Issues_: [#2772](https://github.com/ballerina-platform/ballerina-standard-library/issues/2772) 
+
+## Summary
+The existing replacement functions don't support the following functionalities:
+- There is no way to replace the matches from the particular string index of the given string.
+- There isn't any mechanism to replace the matches with a dynamic value.
+
+So, this proposal improves the replacement functions to support the above functionalities.
+
+## Goals
+
+- Provide a way to support the above functionalities.
+
+## Motivation
+
+This allows users to get the above functionalities when using replacement functions.
+
+## Description
+
+API improvements are:
+
+- Introduce a function to create the new substring to be used to replace the matches. 
+    ```ballerina
+    # A function to be invoked to create the new substring to be used to replace the matches.
+    type ReplacerFunction function(Match 'match) returns string;
+    ```
+
+- Introduce a new type to get the replacement value.  
+    ```ballerina
+    # A function to be invoked to create the new substring or string value to be used to replace the matches.
+    public type Replacement ReplacerFunction|string;
+    ```
+
+- Change the 3rd param of the API with a new type.
```

Comment:
Instead, shall we say "Change the type of `replacement` param from `string `to `Replacement` type."